Add files via upload

gtrtuugii opened the pull request.

gtrtuugii committed:
Add files via upload

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,380 @@
+        else:
+            print(f"HTTP Error: {resp.status}")
+            return None
+    except urllib3.exceptions.HTTPError as e:
```
try block too big, I assume HTTPError is raised by `http.request` so only catch that 1 line:
```py
try:
  resp = http.request(..)
except HttpError:
  pass

if resp.status == 200:
  ...

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,380 @@
+eventbrite_api_token = os.environ.get('EVENTBRITE_API_TOKEN')
+organization_id = os.environ.get('EVENTBRITE_ORGANISATION_ID')
+event_id = os.environ.get('EVENTBRITE_EVENT_ID')
+ticket_class_id = os.environ.get('EVENTBRITE_TICKET_CLASS_ID')
```
Remove event_id, and ticket_class_id, these will be returned by your create event/ticket functions.

robinchew replied:
In fact I want all these `os.environ` calls to be run at the bottom where `if __name == '__main__'` is. Pass those values as arguments to your creation functions.
eg.
```py
def create_function(http, api_token, organisation_id, ...)
```

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,380 @@
+EVENTBRITE_API_BASE_URL = "https://www.eventbriteapi.com/v3/"
+
+# Examples
+event_details = """
```
these should be a dictionaries instead of a string.

gtrtuugii committed:
Update eventbrite.py

Changed function names to snake_case

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,380 @@
+            "error": "HTTP Error",
+            "error_description": str(e),
+            "status_code": e.code
+        }
```
Create a `quick_create_event` function that calls `createEvent`, `createTicketClass` and `publishEvent`:
```py
def quick_create_event(http, organisation_id, title, description, cost_cents=None, publish=True):
  ticket_detail = {
    "ticket_class": {
      "name": "API Test0",
      "free": true,
      "donation": false,
      "quantity_total": "100",
      "minimum_quantity": "1",
      "maximum_quantity": "10",
      "hidden": false,
      "auto_hide": false,
      "auto_hide_before": "",
      "auto_hide_after": "",
      "sales_channels": ["online","atd"],
      "hide_sale_dates": false,
      "delivery_methods": ["electronic"],
      "include_fee": false,
      "sales_start": "2023-11-09T08:00:00Z",
      "sales_end": "2023-11-11T03:00:00Z",
      **({"cost": f"AUD,{cost_cents}"} if cost is not None else {}),
    }
}
```

robinchew replied:
Also need a `date_start` argument. I want to hardcode the venue to be hillview intercultural center https://www.google.com/maps/place/Hillview+Intercultural+Community+Centre/@-32.0090192,115.9003645,17z/data=!3m1!4b1!4m6!3m5!1s0x2a32bdcd5da2c777:0x2fbd272e005ce3d5!8m2!3d-32.0090238!4d115.9029448!16s%2Fg%2F11pckj99n5?entry=ttu

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,380 @@
+            data = resp.data.decode('utf-8')
+            print(data)
+            return json.loads(data)
+        elif resp.status == 400:
```
Remove conditions checking of 200 and 400, assume it's going to be 200. If it's anything but 200 then I want exception to occur.

gtrtuugii committed:
Update eventbrite.py

Resolved comments

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+        "ticket_class": 
+        {
+        "name": "API Test1",
+        "free": "true" if cost_cents is None else "false",
```
put it together with the "cost" key. See below.

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+        "include_fee": "false",
+        "sales_start": "2023-11-09T08:00:00Z",
+        "sales_end": "2023-11-11T03:00:00Z",
+        **({"cost": f"AUD,{cost_cents}"} if cost_cents is not None else {}),
```
```py
        **({"cost": f"AUD,{cost_cents}", "free": "false"} if cost_cents is not None else {'free': "true"}),
```

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+    new_event = create_event(http, organization_id, event_detail_json)
+
+    # Extract the event_id from the response
+    event_id = new_event.get("id")
```
I'm expecting an actual `event_id` with a non-None value, so don't use get:
```py
event_id = new_event['id']
```

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+
+    # Create the ticket class details JSON
+    ticket_detail = {
+        "ticket_class": 
```
Fix your identations.

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+}
+
+paid_ticket_details = {
+        "ticket_class": 
```
fix your indentations

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+    'Content-Type': 'application/json'
+    }
+
+    try:
```
actually why even catch the error? provided the request sent is correct,  we should expect no errors. If for some reason the usual request don't work anymore, I want this exception to occur and not die silently. Remove the try/except.

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+def create_ticket_class(http, event_id, ticket_details):
+    url = f"{EVENTBRITE_API_BASE_URL}events/{event_id}/ticket_classes/"
+    headers = {
+    'Authorization': eventbrite_api_token,
```
Don't treat `eventbrite_api_token` as a global, pass it through the function:
```py
def create_ticket_class(http, eventbrite_api_token, event_id, ticket_details)
```
Apply to rest of the functions.

gtrtuugii committed:
Update eventbrite.py

Resolve comments

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,422 @@
+    data = resp.data.decode('utf-8')
+    return json.loads(data)
+
+def quick_create_event(http, eventbrite_api_token, organization_id, title, description, cost_cents=None, publish=True):
```
missing `date_start` (DateTime object) and `duration_hours` (float) arguments. You can use those 2 values to calculate the end datetime.
```py
from datetime import timedelta
date_start + timedelta(hours=duration_hours)
```

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,422 @@
+            "shareable": False,
+            "invite_only": False,
+            "show_remaining": True,
+            "password": "12345",
```
What's this password for?

gtrtuugii replied:
password | string | Event password used by visitors to access the details of the Event.

robinchew replied:
where does the visitor enter it?

robinchew replied:
Make sure the event is made public.

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,422 @@
+            "online_event": False,
+            "organizer_id": "",
+            "listed": False,
+            "shareable": False,
```
What makes an event not shareable?

gtrtuugii replied:
shareable | boolean | true = Event is shareable, by including social sharing buttons for the Event to Eventbrite applications.

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,422 @@
+            "hide_sale_dates": "false",
+            "delivery_methods": ["electronic"],
+            "include_fee": "false",
+            "sales_start": "2023-11-09T08:00:00Z",
```
How does the ticket sales time affect the event start time?

gtrtuugii replied:
I believe it's the date when tickets start being available for sale.
sales_start | datetime | Date and time that sales for this Ticket Class begin.

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,422 @@
+if __name__ == "__main__":
+
+    # Retrieve the values of an environment variables
+    eventbrite_api_token = os.environ.get('EVENTBRITE_API_TOKEN')
```
these environment variables are mandatory, don't use `.get`

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,422 @@
+    # Create a PoolManager object
+    http = urllib3.PoolManager()
+
+    # quick_create_event(http, eventbrite_api_token, organization_id, "API-TEST-01", "API-TEST-DESCRIPTION", 2000, True)
```
Don't comment this out, perhaps make publish default to False instead. You can even past a mandatory environment variable to control whether to publish or not.

robinchew replied:
For the PUBLISH environment variable, that one you can use `.get`. Default can be either true or false.

gtrtuugii committed:
Resolved comments

1. added start_date and duration_hours
2. removed password and made event public
3. event_id = new_event['id'] was giving me a keyError changed it back for now.
   event_id = new_event.get('id')

robinchew commented on eventbrite.py:
```diff
@@ -0,0 +1,430 @@
+    # Create a PoolManager object
+    http = urllib3.PoolManager()
+
+    quick_create_event(http, eventbrite_api_token, organization_id, "API-TEST-04", "API-TEST-DESCRIPTION-4","2023-11-12T14:00:00Z", 2, 2000, publish)
```
I'm skeptical that the date works if I pass 5pm perth time. eg. "2023-11-12T17:00:00+0800"

robinchew replied:
I want further additional 'timezone' argument to the `quick_create_event`  function, and I want date_start to be timezone UNAWARE. However the quick_create_event code will combine the time unaware `date_start` and `timezone` argument to be timezone aware datetime object, and then you can convert that to UTC if you want, but it looks like the eventbrite API has a timezone field for dates anyway.

Example timezone in IANA format:  'Australia/Perth'

gtrtuugii committed:
Update eventbrite.py

gtrtuugii committed:
Update eventbrite.py

used urllib (built-in) instead of urllib3

gtrtuugii committed:
Update eventbrite.py

remove comments and print statements

robinchew commented on eventbrite.py:
```diff
@@ -87,262 +89,279 @@
+    with urllib.request.urlopen(req) as resp:
+        data = resp.read().decode('utf-8')
+
+        if resp.status != 200:
```
Dedent this, once you've read from resp, there's no need to stay within the context manager:
```py
with urllib.request.urlopen():
  data = resp.read().decode()

if resp.status ! = 200:
```

robinchew commented on eventbrite.py:
```diff
@@ -87,262 +89,279 @@
-        
-    data = resp.data.decode('utf-8')
-    print(data)
+        return print(f"HTTP Error: {resp.status}")
```
print returns None

gtrtuugii committed:
Update eventbrite.py

fix indentation
